.github/workflows/ci.yaml

```diff
@@ -0,0 +1,41 @@
+name: catkin_pkg-ci
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: ['*']
+
+jobs:
+    build:
+      strategy:
+        matrix:
+          os: [ubuntu-latest, macos-latest]
+          python: [3.7, 3.8, 3.9]
+          include:
+          - os: ubuntu-16.04
+            python: 2.7
+          - os: ubuntu-18.04
+            python: 2.7
+          - os: macos-latest
+            python: 2.7
+          - os: ubuntu-16.04
+            python: 3.5
+          - os: ubuntu-18.04
+            python: 3.6
```

Comment:
@cottsay With apologies that I'm experimenting rather than being consistent still I'd love your feedback on this approach to the build matrix. It's the smallest I could make it while still having any coherence. If you think this is more readable and maintainable than the big list of exclusions I'll open PRs to normalize it everywhere.